### ISSUE TITLE
Designing Network Design Spaces

### ISSUE THREAD

**Issue by sile** [sile/readings#147]
URL
-----

https://arxiv.org/abs/2003.13678

Abstract
----------

> In this work, we present a new network design paradigm. Our goal is to help advance the understanding of network design and discover design principles that generalize across settings. Instead of focusing on designing individual network instances, we design network design spaces that parametrize populations of networks. The overall process is analogous to classic manual design of networks, but elevated to the design space level. Using our methodology we explore the structure aspect of network design and arrive at a low-dimensional design space consisting of simple, regular networks that we call RegNet. The core insight of the RegNet parametrization is surprisingly simple: widths and depths of good networks can be explained by a quantized linear function. We analyze the RegNet design space and arrive at interesting findings that do not match the current practice of network design. The RegNet design space provides simple and fast networks that work well across a wide range of flop regimes. Under comparable training settings and flops, the RegNet models outperform the popular EfficientNet models while being up to 5x faster on GPUs.

Table of Contents
--------------------

- (1) Introduction
- (2) Related Work
  - Manual network design
  - Automated network design
  - Network scaling
  - Comparing networks
  - Parameterization
- (3) Design Space Design
  - (3-1) Tools for Design Space Design
  - (3-2) The AnyNet Design Space
  - (3-3) The RegNet Design Space
  - (3-4) Design Space Generalization
- (4) Analyzing the RegNetX Design Space
  - RegNet trends
  - Complexity analysis
  - RegNetX constrained
  - Alternate design choices
  - SE
- (5) Comparison to Existing Networks
  - (5-1) State-of-the-Art Comparison: Mobile Regime
  - (5-2) Standard Baselines Comparison: ResNet(X)t
  - (5-3) State-of-the-Art Comparison: Full Regime
- (6) Conclusion
- (A) Test Set Evaluation
  - Evaluation setup
  - RESNE(X)T comparisons
  - EFFICIENTNET comparisons
- (B) Additional Ablations
  - Fixed depth
  - Fewer stages
  - Inverted Bottleneck
  - Swish vs. ReLU
- (C) Optimization Settings
- (D) Implementation Details
  - Group width compatibility
  - Group width ranges
  - Block types
  - Y block details

BibTeX
------

```bibtex
@misc{radosavovic2020designing,
    title={Designing Network Design Spaces},
    author={Ilija Radosavovic and Raj Prateek Kosaraju and Ross Girshick and Kaiming He and Piotr Dollár},
    year={2020},
    eprint={2003.13678},
    archivePrefix={arXiv},
    primaryClass={cs.CV}
}
```

See Also
----------

- GitHub: https://github.com/facebookresearch/pycls

**Comment by sile** [comment]
> we employ an empirical bootstrap [[5](https://www.amazon.co.jp/dp/B07YYMMB7P)] to estimate the likely range in which the best models fall.

**Comment by sile** [comment]
> To summarize: (1) we generate distributions of models obtained by sampling and training n models from a design space, (2) we compute and plot error EDFs to summarize design space quality, (3) we visualize various properties of a design space and use an empirical bootstrap to gain insight, and (4) we use these insights to refine the design space.
